modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderLocalServiceTest.java

```diff
@@ -0,0 +1,326 @@
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				null, RandomTestUtil.randomString()));
+
+		AssertUtils.assertFailure(
+			ObjectFolderNameException.MustBeLessThan41Characters.class,
+			"Name must be less than 41 characters",
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				RandomTestUtil.randomString(42)));
+
+		AssertUtils.assertFailure(
+			ObjectFolderNameException.MustNotBeDuplicate.class,
+			"Duplicate name " + _uncategorizedObjectFolder.getName(),
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				_uncategorizedObjectFolder.getName()));
+
+		AssertUtils.assertFailure(
+			ObjectFolderNameException.MustNotBeNull.class, "Name is null",
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				null));
+
+		AssertUtils.assertFailure(
+			ObjectFolderNameException.MustOnlyContainLettersAndDigits.class,
+			"Name must only contain letters and digits",
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"Abl e"));
+
+		AssertUtils.assertFailure(
+			ObjectFolderNameException.MustOnlyContainLettersAndDigits.class,
+			"Name must only contain letters and digits",
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"Abl-e"));
+
+		String externalReferenceCode = RandomTestUtil.randomString();
+		Map<Locale, String> labelMap = LocalizedMapUtil.getLocalizedMap(
+			RandomTestUtil.randomString());
+		String name = RandomTestUtil.randomString();
+
+		ObjectFolder objectFolder = _objectFolderLocalService.addObjectFolder(
+			externalReferenceCode, TestPropsValues.getUserId(), labelMap, name);
+
+		_validateObjectFolder(
+			externalReferenceCode, labelMap, name, objectFolder);
+
+		Assert.assertEquals(
+			1,
+			_resourcePermissionLocalService.getResourcePermissionsCount(
+				objectFolder.getCompanyId(), ObjectFolder.class.getName(),
+				ResourceConstants.SCOPE_INDIVIDUAL,
+				String.valueOf(objectFolder.getObjectFolderId())));
+
+		_objectFolderLocalService.deleteObjectFolder(objectFolder);
+	}
+
+	@Test
+	public void testDeleteCompanyObjectFolders() throws Exception {
+		String originalName = PrincipalThreadLocal.getName();
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		Company company = CompanyTestUtil.addCompany();
```

Comment:
I believe company initialization should come before originalName. Do you agree?

Reply:
Hey @seltong 

`originalName` and `originalPermissionChecker` always come before the created company in tests that I found.